blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -135,11 +143,33 @@ private boolean isReplayable(ReplayAction replayAction) {
         return replayAction != null && replayAction.isRebuildEnabled();
     }
 
+    static final long PIPELINE_NODE_CONTAINER_CACHE_MAX_SIZE = Long.getLong("PIPELINE_NODE_CONTAINER_CACHE_MAX_SIZE", 10000);
+
+    private static Cache<String, PipelineNodeContainerImpl> PIPELINE_NODE_CONTAINER_LOADING_CACHE = Caffeine.newBuilder()
```

Comment:
For future readers, nice to document what the key is. Some `.link.href` which would be, what, `job/mbp/job/main/123/`? Or `/blue/something`?

(Normally a `WorkflowRun` would be keyed by its `externalizableId`.)

Reply:
done https://github.com/jenkinsci/blueocean-plugin/commit/ae3da2701919e5d6c4dfa1fecd054754f247e2d8